HardwareID/HardwareID.cs

```diff
@@ -0,0 +1,256 @@
+        /// <param name="seed">seed to add to system info for hash</param>
+        /// <returns>string hash of _data + seed</returns>
+        public static string UniqueID(string seed)
+        {
+            return GetHash($"{_data}\nSeed >> {seed}");
+        }
+        #endregion
+
+        #region helper methods
+        /// <summary>
+        /// Run system process `systeminfo` to get data
+        /// </summary>
+        private static void GetSystemInfo()
+        {
+            if (Environment.OSVersion.ToString().StartsWith("Microsoft"))
+            {
+                // on Windows
+                Process p = new Process();
+                p.StartInfo.UseShellExecute = false;
+                p.StartInfo.CreateNoWindow = true;
+                p.StartInfo.RedirectStandardOutput = true;
+                p.StartInfo.FileName = "systeminfo";
+                p.Start();
+                string output = p.StandardOutput.ReadToEnd();
+                p.WaitForExit();
+
+                // split the output based on the system's new line charecter(s) - don't use any empty splits
+                foreach (var line in output.Split(Environment.NewLine.ToCharArray(), StringSplitOptions.RemoveEmptyEntries))
+                {
+                    var workingLine = line.Trim();
+                    var match = systemInfoFields.FirstOrDefault(x => workingLine.StartsWith(x.Key));
+                    if (!match.Equals(default(KeyValuePair<string, string>)))
+                    {
+                        var rest = workingLine.Substring(match.Key.Length).Trim();
+                        data[match.Value] = rest;
+                    }
+                }
+            }
+            else
+            {
+                Process p = new Process();
+                p.StartInfo.UseShellExecute = false;
+                p.StartInfo.CreateNoWindow = true;
+                p.StartInfo.RedirectStandardOutput = true;
+                p.StartInfo.FileName = "uname";
+                p.StartInfo.Arguments = "-v";
+                p.Start();
+                string output = p.StandardOutput.ReadToEnd();
+                p.WaitForExit();
+                data[nameof(UName)] = output;
+            }
+        }
+
+        /// <summary>
+        ///  Get hash of string (translated to bytes)
+        /// hashing code from http://forum.codecall.net/topic/78149-c-tutorial-generating-a-unique-hardware-id/
+        /// </summary>
+        /// <param name="s">the value to hash</param>
+        /// <returns>Hexidecimal formatted hash string</returns>
+        private static string GetHash(string s)
+        {
+            // Initialize a new MD5 Crypto Service Provider in order to generate a hash
+            MD5 sec = new MD5CryptoServiceProvider();
+
+            // Grab the bytes of the variable 's'
+            byte[] bt = Encoding.ASCII.GetBytes(s);
+
+            // Grab the Hexadecimal value of the MD5 hash
+            return GetHexString(sec.ComputeHash(bt));
```

Comment:
add a comment explaining why string.GetHashCode() wasn't good enough for your purposes.